src/IIIF/IIIF/Serialisation/SourceConverter.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using IIIF.Presentation.V3;
+using IIIF.Presentation.V3.Annotation;
+using IIIF.Presentation.V3.Content;
+using IIIF.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace IIIF.Serialisation;
+
+public class SourceConverter : JsonConverter<IPaintable>
```

Comment:
There's a lot of commonality between this and `TargetConverter` - can they share logic? Stuff like `IsSimpleCanvas()` etc